fix: Add `continue-on-error` to codecov action step

mafredri opened the pull request:
Hopefully works around: https://github.com/codecov/codecov-action/issues/788

Might bypass the failure, and makes sense not to trust the action with managing failure if we want to ignore it anyway.

mafredri committed:
fix: Add `continue-on-error` to codecov action step

Hopefully fixes: https://github.com/codecov/codecov-action/issues/788

dwahler commented:
Might want to also remove the `fail_ci_if_error` parameter, so that the codecov step itself returns an error status and is visibly marked as failing, even if it doesn't cause the rest of the job to fail.

mafredri committed:
chore: Add comment explaining codecov action failure

mafredri commented:
@dwahler actually, I'm not brave enough to set `fail_ci_if_error: true` since I don't know by what mechanism it's going to fail the action. As-such, I just cleaned it up a bit and added a comment explaining it.
